sklearn/svm/tests/test_svm.py

```diff
@@ -306,14 +309,39 @@ def test_decision_function():
     assert_array_almost_equal(clf.decision_function(X), expected, 2)
 
     # kernel binary:
-    clf = svm.SVC(kernel='rbf', gamma=1)
+    clf = svm.SVC(kernel='rbf', gamma=1, compact_decision_function=False)
     clf.fit(X, Y)
-    
+
     rbfs = rbf_kernel(X, clf.support_vectors_, gamma=clf.gamma)
     dec = np.dot(rbfs, clf.dual_coef_.T) + clf.intercept_
     assert_array_almost_equal(dec.ravel(), clf.decision_function(X))
 
 
+def test_compact_decision_function():
+    # check that compact_decision_function=True gives
+    # correct shape and is consistent with predict
+    clf = svm.SVC(kernel='linear', C=0.1,
+                  compact_decision_function=True).fit(iris.data, iris.target)
+    dec = clf.decision_function(iris.data)
+    assert_array_equal(clf.predict(iris.data), np.argmax(dec, axis=1))
+    assert_equal(dec.shape, (len(iris.data), 3))
+
+    # with five classes:
+    X, y = make_blobs(n_samples=80, centers=5, random_state=0)
+    X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=0)
+
+    clf = svm.SVC(kernel='linear', C=0.1,
+                  compact_decision_function=True).fit(X_train, y_train)
+    dec = clf.decision_function(X_test)
+    assert_array_equal(clf.predict(X_test), np.argmax(dec, axis=1))
+    assert_equal(dec.shape, (len(X_test), 5))
+
```

Comment:
I think we need one more similar test on a binary classification problem with `compact_decision_function=True`.